pd-client/client.go

```diff
@@ -622,6 +628,79 @@ func (c *client) GetStore(ctx context.Context, storeID uint64) (*metapb.Store, e
 	return store, nil
 }
 
+func (c *client) GetUserKV(ctx context.Context, key string) (string, error) {
+	if span := opentracing.SpanFromContext(ctx); span != nil {
+		span = opentracing.StartSpan("pdclient.GetUserKV", opentracing.ChildOf(span.Context()))
+		defer span.Finish()
+	}
+	start := time.Now()
+	defer func() { cmdDuration.WithLabelValues("get_user_key").Observe(time.Since(start).Seconds()) }()
+
+	ctx, cancel := context.WithTimeout(ctx, pdTimeout)
+	resp, err := c.leaderClient().GetUserKV(ctx, &pdpb.GetUserKVRequest{
+		Header: c.requestHeader(),
+		Key:    key,
+	})
+	requestDuration.WithLabelValues("get_user_key").Observe(time.Since(start).Seconds())
+	cancel()
+
+	if err != nil {
+		cmdFailedDuration.WithLabelValues("get_user_key").Observe(time.Since(start).Seconds())
+		c.scheduleCheckLeader()
+		return "", errors.Trace(err)
+	}
+	return resp.GetValue(), nil
+}
+
+func (c *client) PutUserKV(ctx context.Context, key string, value string) error {
+	if span := opentracing.SpanFromContext(ctx); span != nil {
+		span = opentracing.StartSpan("pdclient.PutUserKV", opentracing.ChildOf(span.Context()))
+		defer span.Finish()
+	}
+	start := time.Now()
+	defer func() { cmdDuration.WithLabelValues("put_user_key").Observe(time.Since(start).Seconds()) }()
+
+	ctx, cancel := context.WithTimeout(ctx, pdTimeout)
+	_, err := c.leaderClient().PutUserKV(ctx, &pdpb.PutUserKVRequest{
```

Comment:
maybe we should not just ignore the resp, resp.Body.Close() must be called otherwise there will be a leaking. see https://groups.google.com/forum/#!topic/golang-nuts/148riho42sU